checker.cpp

```diff
@@ -1,22 +1,53 @@
-#include <assert.h>
-#include <iostream>
-using namespace std;
-
-bool batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
-    cout << "Temperature out of range!\n";
-    return false;
-  } else if(soc < 20 || soc > 80) {
-    cout << "State of Charge out of range!\n";
-    return false;
-  } else if(chargeRate > 0.8) {
-    cout << "Charge Rate out of range!\n";
-    return false;
+#include "checker.hpp"
+
+bool bBatteryIsOk(float temperature, float soc, float chargeRate) 
+{
+  bool temperatureCondition = true;
+  bool socCondition = true;
+  bool chargeRateCondition = true;
+  bool batteryCondition = false;
+
+  temperatureCondition = bTemperatureIsOk(temperature);
+  socCondition = bSocIsOk(soc);
+  chargeRateCondition = bChargeRateIsOk(chargeRate);
+  
+  batteryCondition = (temperatureCondition && socCondition && chargeRateCondition);
+  
+  return batteryCondition;
+}
+
+bool bTemperatureIsOk(float temperature)
+{
+  bool retTemperatureCondition = true;
+  if(temperature < 0 || temperature > 45) 
```

Comment:
usage of hard coded values will lead to issues during maintenance. if it gets used in many places and then a change in one place needs to be consistent with all others. There is a scope for improvement here.